wire-library/wire-runtime-swift/src/main/swift/propertyWrappers/JSONEnum.swift

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2021 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+
+/**
+ Converts enums to/from their string equivalent when serializing to/from JSON.
+ This matches the Proto3 JSON spec: https://developers.google.com/protocol-buffers/docs/proto3#json
+ */
+@propertyWrapper
+public struct JSONEnum<T : CaseIterable & Hashable & RawRepresentable> : Codable, Hashable where T.RawValue == UInt32 {
```

Comment:
I will be applying this pattern of having 3 property wrappers to the current JSONString.
Of note, I will mark `JSONString` as deprecated and create new property wrappers.

Reply:
I'm tempted to say a better name for this would be `ProtoEnumEncoded`.

I'm also tempted to suggest that we add `LosslessStringConvertible` to proto enums